src/Services/API/Project/ProjectService.API/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,121 @@
+﻿using ProjectService.Contract.DataTransfer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using EnterpriseAssistant.DataAccess;
+using Microsoft.EntityFrameworkCore;
+using EnterpriseAssistant.DataAccess.Entities;
+using Mapster;
+
+namespace ProjectService.API.Controllers;
+
+[AllowAnonymous]
+[ApiController]
+[Route("api/project")]
+public class ProjectController : ControllerBase
+{
+    private readonly EnterpriseAssistantDbContext _context;
+
+    public ProjectController(EnterpriseAssistantDbContext context)
+    {
+        _context = context;
+    }
+
+    /*
+    [HttpPost]
+    [SwaggerOperation(Summary ="Create project", Description = "Create a project")]
+
+    public async Task<ActionResult<ProjectCreateDto>> CreateProject([FromBody] ProjectCreateDto project)
+    {
+        var result = await _mediator.Send(new CreateProject(project));
+        return Ok(result);
+    }
+
+
+    [HttpGet("{projectId:long}")]
+    [SwaggerOperation(Summary = "Get prject by Id", Description = "Get project")]
+    public async Task<ActionResult<IEnumerable<ProjectDto>>> GetProjectById(
+        [Range(1, long.MaxValue), FromRoute] long projectId)
+    {
+        var result = await _mediator.Send(new GetProjectById(projectId));
+
+        return result.Match<ActionResult>(Ok, x => NotFound());
+    }
+
+    [HttpDelete("{projectId:long}")]
+     [SwaggerOperation(Summary = "Delete project by id", Description = "delete project")]
+      public async Task<IActionResult> ProjectDelete(long id)
+     {
+
+     }
+    */
+
+    [HttpPost]
+    [SwaggerOperation(Summary = "Add project", Description = "add project")]
+    public async Task<ActionResult<Project>> PostProject(ProjectCreateDto request)
+    {
+        var project = request.Adapt<Project>();
+        _context.Projects.Add(project);
+        await _context.SaveChangesAsync();
+
+        return CreatedAtAction(nameof(PostProject), new { id = project.Id }, project);
+    }
+
+    [HttpDelete("{id}")]
+    [SwaggerOperation(Summary = "Delete project by id", Description = "delete project")]
+    public async Task<IActionResult> DeleteProject(int id)
+    {
+        var project = await _context.Projects.FindAsync(id);
+        if (project == null)
+        {
+            return NotFound();
+        }
+
+        _context.Projects.Remove(project);
```

Comment:
use soft delete instead of hard, just set IsSoftDeleted to true and then update the project

Reply:
Like this?
![1](https://user-images.githubusercontent.com/62302984/172028194-c9e33909-9418-4695-8aad-41041a6099f1.png)

Reply:
I'm not sure, need to check it myself

Reply:
![image](https://user-images.githubusercontent.com/46058617/172419653-07109902-440b-4fa2-8f14-43031c4490a2.png)
use this if you update one property and Update for updating entire entity